### PR TITLE
add python solution for rest stops (810)

### PR DESCRIPTION
**Pull request by jjliewie**
_If any of the below doesn't apply to this Pull Request, mark the checkbox as done._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions), which includes the following: 
  - I understand that if it is clear that I have not attempted to follow these guidelines (ex. if I have not used tabs to indent), my PR will be closed.
  - If changes are requested, I will re-request the review after making them.

### CODE REVIEW

File: solutions/usaco-810.mdx [hunk 1]
```diff
@@ -161,4 +161,32 @@ public class RestStops {
...
 
+<PySection>
+
+```python
+import sys
+
+sys.stdin = open("reststops.in", "r")
+sys.stdout = open("reststops.out", "w")
+
+input = sys.stdin.readline
```
Comment (ryanchou-dev) [inline solutions/usaco-810.mdx:172]: .

File: solutions/usaco-810.mdx [hunk 2]
```diff
@@ -161,4 +161,32 @@ public class RestStops {
...
+
+x, c = [0] * 100005, [0] * 100005
```
Comment (ryanchou-dev) [inline solutions/usaco-810.mdx:174]: make 1e5+5 a constant

or just define underneath `n`

File: solutions/usaco-810.mdx (hunk 1 above)
Comment (jjliewie) [reply solutions/usaco-810.mdx:172]: ookay

File: solutions/usaco-810.mdx (hunk 2 above)
Comment (jjliewie) [reply solutions/usaco-810.mdx]: done

**Review by ryanchou-dev — APPROVED** [review]
⭐

File: solutions/usaco-810.mdx [hunk 3]
```diff
@@ -161,4 +161,26 @@ public class RestStops {
...
+	
+	x, c = [0] * (n + 1), [0] * (n + 1)
+	
+	for i in range(1, n + 1):
+		x[i], c[i] = map(int, read.readline().split())
+
+ans = 0
+
+for i in range(n, 0, -1):
+	c[i] = max(c[i], c[i + 1])
```
Comment (envyaims) [inline solutions/usaco-810.mdx:178]: list index out of range lol

**Review by envyaims — CHANGES_REQUESTED** [review]
:)

File: solutions/usaco-810.mdx (hunk 3 above)
Comment (jjliewie) [reply solutions/usaco-810.mdx:178]: no way let me check... all the solutions should work (they worked in the beginning)

Comment (jjliewie) [reply solutions/usaco-810.mdx:178]: @envyaims its not bc the size of c is n + 1 (refer to line 170)

Comment (envyaims) [reply solutions/usaco-810.mdx:178]: try submitting it

Comment (jjliewie) [reply solutions/usaco-810.mdx:178]: oo changed it should work now @/ whoever told me change my solution before u were wrong ig.

File: solutions/usaco-810.mdx [hunk 4]
```diff
@@ -161,4 +161,26 @@ public class RestStops {
 
 </JavaSection>
 
+<PySection>
+
+```python
+x, c = [0] * 100005, [0] * 100005
```
Comment (envyaims) [inline solutions/usaco-810.mdx:167]: make 1e5 a constant pls

**Review by envyaims — CHANGES_REQUESTED** [review]

**Review by nwatx — APPROVED** [review]

File: solutions/usaco-810.mdx (hunk 4 above)
Comment (Aadit-Ambadkar) [reply solutions/usaco-810.mdx:167]:
```suggestion
MX = 100005
x, c = [0] * MX, [0] * MX
```

**Review by Aadit-Ambadkar — APPROVED** [review]
looks pretty good

File: solutions/usaco-810.mdx [hunk 5]
```diff
@@ -161,4 +161,27 @@ public class RestStops {
 
 </JavaSection>
 
+<PySection>
+
+```python
+MX = 100005
```
Comment (SansPapyrus683) [inline solutions/usaco-810.mdx:167]: remove the +5

File: solutions/usaco-810.mdx [hunk 6]
```diff
@@ -161,4 +161,27 @@ public class RestStops {
...
+x, c = [0] * MX, [0] * MX
```
Comment (SansPapyrus683) [inline solutions/usaco-810.mdx:168]: wait please just use dynamic arrays
it isn't even that hard
just go to the `with` block
```py
x, c = [0] * (whatever), [0] * (whatever)
```

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]
how did i not review this

File: solutions/usaco-810.mdx (hunk 6 above)
Comment (SansPapyrus683) [reply solutions/usaco-810.mdx:168]: bump

**Pull request closed without merging**